app/src/main/java/com/example/musicapp/ui/screenplayer/FragmentPlayer.kt

```diff
@@ -0,0 +1,178 @@
+package com.example.musicapp.ui.screenplayer
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.SeekBar
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.musicapp.R
+import com.example.musicapp.data.model.Song
+import com.example.musicapp.data.repository.SongRepository
+import com.example.musicapp.data.source.local.SongLocalDataSource
+import com.example.musicapp.notifications.*
+import com.example.musicapp.service.MusicService
+import com.example.musicapp.ui.adapter.SongAdapter
+import com.example.musicapp.utils.OnItemRecyclerViewClickListener
+import com.example.musicapp.utils.SimpleTimeFormat
+
+import kotlinx.android.synthetic.main.fragment_player.*
+
+const val EXCESS_TIME = 500
+
+class FragmentPlayer(val service: MusicService) : Fragment(), PlayerContract.View,
+    View.OnClickListener, OnItemRecyclerViewClickListener, SeekBar.OnSeekBarChangeListener {
+    lateinit var presenter: PlayerPresenter
+    lateinit var notification: MusicNotification
+    lateinit var song: Song
+    val songAdapter: SongAdapter = SongAdapter()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_player, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initRecyclerView()
+        initData()
+        initNotify()
+        initListener()
+
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        activity?.unregisterReceiver(broadcastReceiver)
+        notification.cancelNotification()
+
+    }
+
+    override fun onClick(v: View?) {
+        when (v?.id) {
+            R.id.buttonPlay -> {
+                presenter.handlePlaySong(null)
+            }
+            R.id.buttonNext -> presenter.handleNextSong()
+            R.id.buttonPrev -> presenter.handlePrevSong()
+        }
+    }
+
+
+    override fun loadSuccess(list: List<Song>) {
```

Comment:
đặt tên hàm cho hành động nó thực hiện 
```suggestion
    override fun showSongs(list: List<Song>) {
```